arm/authorization/client.go

```diff
@@ -0,0 +1,759 @@
+		autorest.WithErrorUnlessOK(),
+		autorest.ByClosing())
+
+	return
+}
+
+func (client *ManagementLocksClient) NewDeleteAtResourceLevelRequest(resourceGroupName string, resourceProviderNamespace string, parentResourcePath string, resourceType string, resourceName string, lockName string) (*http.Request, error) {
+
+	pathParameters := map[string]interface{}{
+		"lockName":                  url.QueryEscape(lockName),
+		"parentResourcePath":        parentResourcePath,
+		"resourceGroupName":         url.QueryEscape(resourceGroupName),
+		"resourceName":              url.QueryEscape(resourceName),
+		"resourceProviderNamespace": url.QueryEscape(resourceProviderNamespace),
+		"resourceType":              resourceType,
+		"subscriptionId":            url.QueryEscape(client.SubscriptionId),
+	}
+
+	queryParameters := map[string]interface{}{
+		"api-version": ApiVersion,
+	}
+
+	return autorest.DecoratePreparer(
+		client.DeleteAtResourceLevelRequestPreparer(),
+		autorest.WithPathParameters(pathParameters),
+		autorest.WithQueryParameters(queryParameters)).Prepare(&http.Request{})
+}
+
+func (client *ManagementLocksClient) DeleteAtResourceLevelRequestPreparer() autorest.Preparer {
+	return autorest.CreatePreparer(
+		autorest.AsJSON(),
+		autorest.AsDelete(),
+		autorest.WithBaseURL(client.BaseUri),
+		autorest.WithPath("/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{parentResourcePath}/{resourceType}/{resourceName}/providers/Microsoft.Authorization/locks/{lockName}"))
+}
+
+// CreateOrUpdateAtSubscriptionLevel create or update a management lock at the
+// subscription level.;
+//
+// lockName is the name of lock. parameters is the management lock parameters.
+func (client *ManagementLocksClient) CreateOrUpdateAtSubscriptionLevel(lockName string, parameters ManagementLock) (result ManagementLock, err error) {
+
+	req, err := client.NewCreateOrUpdateAtSubscriptionLevelRequest(lockName, parameters)
+	if err != nil {
+		return result, fmt.Errorf("authorization: Failure preparing ManagementLocksClient.CreateOrUpdateAtSubscriptionLevel request (%v)", err)
+	}
+
+	req, err = autorest.Prepare(
+		req,
+		client.WithAuthorization(),
+		client.WithInspection())
+	if err != nil {
+		return result, fmt.Errorf("authorization: Failure sending ManagementLocksClient.CreateOrUpdateAtSubscriptionLevel request (%v)", err)
+	}
+
+	resp, err := autorest.SendWithSender(client, req)
+	if err != nil {
+		return result, fmt.Errorf("authorization: Failure sending ManagementLocksClient.CreateOrUpdateAtSubscriptionLevel request (%v)", err)
+	}
+
+	result = ManagementLock{}
+	err = autorest.Respond(
+		resp,
+		client.ByInspecting(),
+		autorest.WithErrorUnlessOK(),
+		autorest.ByUnmarshallingJSON(&result),
+		autorest.ByClosing())
+
+	return result, err
+}
+
+func (client *ManagementLocksClient) NewCreateOrUpdateAtSubscriptionLevelRequest(lockName string, parameters ManagementLock) (*http.Request, error) {
```

Comment:
So the helper equivalent of this (`CreateOrUpdateAtSubscriptionLevel` method) has some godoc above, and the parameters are all the same. Maybe add godoc for these methods, too? It's obvious, but it'll help if we add golint checks to the repo. (currently "method is missing documentation" checks are suppressed).